draft-ietf-masque-h3-datagram.md

```diff
@@ -623,69 +339,24 @@ recent draft version from the intersection set. This ensures that both
 endpoints negotiate the same draft version.
 
 
-# The Sec-Use-Datagram-Contexts HTTP Header {#context-hdr}
+# The Sec-Capsule-Protocol Header
```

Comment:
Why does this have a Sec- prefix?  Why shouldn't Javascript be allowed to set this header on requests it generates?  It seems safe to me.

Reply:
That sounds dangerous, if we allow JavaScript to send this we're adding a burden on every new capsule type to consider the implications of that. I'd rather disallow it to limit risk and avoid potential issues. If someone has a use-case that requires this we could reconsider that choice.

Reply:
I know that we just shove "Sec-" on everything nowadays, but I think we need to meet a higher bar than "sounds dangerous".  Do you have a concrete example?

Let's say that I did a POST using fetch() with this new field attached.  I do that over h2 and a CDN upgrades to h3 toward the origin, splitting out apparent "datagrams" into frames on that leg.  The server might still accept the request, but discard the datagrams.  That means that the CDN and origin server see a different payload.  Maybe that's something that can be exploited.

But then, what is special about that being dependent on JavaScript?  I don't think that it works as an attack. Fetch will preflight the request and will not add the header unless it is specifically allowed by the server in a preflight, see [CORS-unsafe request-header names](https://fetch.spec.whatwg.org/#cors-unsafe-request-header-names).

Reply:
That's fair. You understand CORS better than I do. After thinking about this some more I have no strong opinion here so I'm happy to go with the consensus of the design team and/or WG on whether we should start with "Sec-" or not.

Reply:
I've removed the Sec- prefix.